routing.md

```diff
@@ -1,4 +1,277 @@
+
+![scenario 6: herd hosting](collateral/routing-scenario-6.png)
+
+This scenario is one that highlights an __external mediator__--so-called because the mediator lives outside the sovereign domain of the final recipient.
+
+#### Scenario 7: intra-domain dispatch
+
+Now let's subtract agents-r-us. We're back to Bob's cloud agent listening directly at http://bob.com/agent. However, let's say that Alice has a different goal--now she wants to talk to the edge agent running on Bob's mobile device. This agent doesn't have a permanent IP address, so Bob uses his own cloud agent as a mediator. He tells Alice that his mobile device agent can only be reached via his cloud agent.
+
+![routing scenario 7: intra-domain dispatch](collateral/routing-scenario-7.png)
+
+Once again, this causes Alice to modify her behavior. Again, she wraps her encrypted message. The inner message is enclosed in an outer envelope, and the outer envelope is passed to the mediator.
+
+This scenario highlights an __internal mediator__. Internal and external mediators introduce similar features and similar constraints; the relevant difference is that internal mediators live within the sovereign domain of the recipient, and may thus be worthy of greater trust.
+
+#### Scenario 8: double mediation
+
+Now let's combine. Bob's cloud agent is hosted at agents-r-us, AND Alice wants to reach Bob's mobile:
+
+![routing scenario 8: double mediation](collateral/routing-scenario-8.png)
+
+This is a common pattern with HTTP-based cloud agents plus mobile edge agents, which is the most common deployment pattern we expect for many users of self-sovereign identity. Note that the properties of the agency and the routing agent are not particularly special--they are just an external and an internal mediator, respectively.
+
+### Remember Routes are One-Way (not Duplex)
+
+In all of this discussion, note that we are analyzing *only* a flow from Alice to Bob. How Bob gets a message back to Alice is a completely separate question. Just because Carl, Darla, Emil, Francis, and Agents-R-Us may be involved in how messages flow from Alice to Bob, does not mean they are involved in flow the opposite direction.
+
+Note how this breaks the simple assumptions of pure request-response technologies like HTTP, that assume the channel in (request) is also the channel out (response). [Duplex](https://en.wikipedia.org/wiki/Duplex_%28telecommunications%29) request-response can be modeled with DIDComm, but doing so requires support that may not always be available, plus cooperative behavior governed by the [`~thread`](https://github.com/hyperledger/aries-rfcs/blob/master/features/0008-message-id-and-threading/README.md) decorator.
+
+### Routing Protocol
```

Comment:
Would “Forwarding Protocol” title fit better? This section does not explain the routing protocol itself, how the sender or the mediator know where to send the message, but it describes the protocol of sending the forward message once the next node has been selected.